build-tools/packages/build-cli/src/commands/release/report-unreleased.ts

```diff
@@ -5,291 +5,104 @@
- * @param gitHubUrl - GitHub Release API URL.
- * @param version - The version number.
- */
-async function generateReleaseReportForUnreleasedVersions(
-	gitHubUrl: string,
-	version: string,
-	log?: Logger,
-): Promise<PackageVersionList | undefined> {
-	try {
-		const releasesResponse = await fetch(gitHubUrl);
 		// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-		const releases = await releasesResponse.json();
-
-		let manifestAsset;
+		const manifestFile: PackageVersionList = JSON.parse(manifestData);
 
-		// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-		for (const asset of releases[0].assets) {
-			/**
-			 * Only `caret` manifest file is required by partner repository.
-			 * dev/prerelease versions are mapped to a single version instead of ranges but other packages such as common-utils and common-definitions
-			 * required caret versions which simple manifest files do not provide.
-			 * Example of simple manifest file:
-			 * ```json
-			 * {
-			 * 	"@fluidframework/cell": "2.0.0-internal.6.1.1",
-			 *	"@fluidframework/common-definitions": "0.20.1",
-			 *	"@fluidframework/common-utils": "1.1.1",
-			 * }
-			 * ```
-			 * Example of caret manifest file:
-			 * ```json
-			 * {
-			 * 	"@fluidframework/cell": ">=2.0.0-internal.6.1.1 <2.0.0-internal.7.0.0",
-			 *	"@fluidframework/common-definitions": "^0.20.1",
-			 *	"@fluidframework/common-utils": "^1.1.1",
-			 * }
-			 * ```
-			 */
-			/* eslint-disable @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-argument */
-			const includesCaretJson: boolean = asset.name.includes(".caret.json");
-			if (includesCaretJson) {
-				manifestAsset = asset;
-				break;
+		for (const key of Object.keys(manifestFile)) {
+			if (ignorePackageList.has(key)) {
+				continue;
 			}
-		}
 
-		if (Object.keys(manifestAsset).length > 0) {
-			const manifest_url_caret = manifestAsset.browser_download_url;
-			const manifestResponse = await fetch(manifest_url_caret);
-			const manifestData = await manifestResponse.buffer();
-
-			const manifestFile: PackageVersionList = JSON.parse(manifestData.toString());
-
-			for (const key of Object.keys(manifestFile)) {
-				if (isInternalVersionRange(manifestFile[key], true)) {
-					manifestFile[key] = version;
-				}
+			if (
+				isInternalVersionRange(manifestFile[key], true) ||
+				manifestFile[key].includes("-rc.")
+			) {
+				manifestFile[key] = devVersion;
 			}
-
-			return manifestFile;
 		}
 
-		return undefined;
-		/* eslint-enable @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-argument */
-	} catch (error) {
-		log?.errorLog("Error generating manifest object:", error);
-		return undefined;
-	}
-}
+		const currentDate = new Date().toISOString().slice(0, 10);
```

Comment:
toISOString() returns a string including date, time, and timezone. By using .slice(0, 10), I'm extracting only the date part, keeping the first 10 characters. The date is needed without the time and timezone information.

Reply:
Suggest using [date-fns](https://date-fns.org/) to deal with this. It will be clearer and more self-documenting. We already have the dependency I think.